element-handles-explainer.md

```diff
@@ -0,0 +1,418 @@
+
+For example:
+
+```html
+<label for="x-input::the-input">Example</label>
+<x-input id="x-input">
+  #shadowRoot
+  | <input id="the-input" exportid />
+  </template>
+</x-input>
+```
+
+**Pros**
+
+* Doesn't introduce another "id-like" attribute (`handle`).
+* Still allows explicit opt-in to exporting elements by ID.
+
+**Cons**
+
+* Exported IDs don't work quite the same way as normal IDs; for example `getElementById('x-input::the-input')` can't return the actual `'the-input'` element, since that would break encapsulation. 
+  * The same fact is true for handles, but having a different name "handle" might make it clearer that it is different?
+* Does not support renaming the exported ID. Theoretically this could be supported with something like `exportid="renamed-id"`, but then the `exportid` property would function almost identically to the `handle` property in this proposal.
+   * That said, it may be worth considering renaming `handle` to `exportid` in this proposal?
+* Does not support having multiple exported IDs for a single element (this is a "nice to have" feature of handles, but certainly not "must have").
+* "Only" saves a little bit of typing. The equivalent of `<div id="some-id" exportid>` with handles is `<div id="some-id" handle="some-id">`.
+
+### Could the syntax for `importhandles` be improved?
+
+Is it confusing that `importhandles` maps handle names to ids, unlike `exporthandles`, which maps handle names to handle names? It may be worthwhile considering alternative APIs for this. 
+
+Brainstorming a few possible alternatives:
+* Rename the attribute to "definehandles" or some other name that doesn't imply it is similar to "exporthandles"
+   * `definehandles="handle-name: my-id::handle(my-handle)"`
+* Abandon the "handle" naming and call it something like "importids".
+   * On the host element: `importids="my-labelledby: my-id::handle(my-handle)`
+   * Inside the shadow tree: referred using e.g. `aria-labelledby=":host::importid(my-labelledby)"`
+* Use custom attributes for each imported handle, rather than combining them all in one attribute. Similar to `data-*` attributes, these could all have a predefined prefix like `importid-*`:
+    ```html
+   <label id="the-label">Example Label</label>
+   <x-input importid-my-labelledby="the-label">
+     #shadowRoot
+     | <input aria-labelledby=":host::importid(my-labelledby)" type="text" />
+   </x-input>
+   ```
+
+### Does `importhandles` need to support attributes with multiple IDs?
+
+The current syntax for `importhandles` doesn't support referencing multiple IDs from a single handle. This may be important for attributes like `aria-labelledby`, which can reference more than one element. For example, it won't work to write `importhandles="my-labelledby: first-label second-label"`. 
```

Comment:
Is this really an issue? Couldn't you import two handles, and then set your `aria-labelledby` to point to both of them (space-separated)?

```html
<label id="label1">Label 1</label>
<label id="label2">Label 2</label>
<x-input importhandles="my-label1: label1, my-label2: label2">
  #shadowRoot
  | <input aria-labelledby=":host::handle(my-label1) :host::handle(my-label2)" type="text">
</x-input>
```

Reply:
This limitation would mainly affect component library authors, who want to create general-purpose web components that have parity with built-in controls. For example, it would not be possible to create a `<fancy-input>` that supports having an arbitrary number of elements for its `aria-labelledby`, in the way that the built-in `<input>` does.

I'll add that to the document. It's not a huge limitation, but may be worth taking into account for potential improvements to the `importhandles` syntax from what I have so far.